src/snap/metric.cc

```diff
@@ -263,3 +263,87 @@ void Metric::Metric::set_last_advert_tm(system_clock::time_point tp) {
     tm->set_sec(nanos / std::nano::den);
     tm->set_nsec(nanos % std::nano::den);
 }
+
+Namespace::Namespace(std::vector<std::string> ns){
+    
+    for (auto &string_iterator : ns){
+        this->namespace_elements.push_back(NamespaceElement(string_iterator));
+    }
+}
+Namespace::Namespace(){}
+
+Namespace::~Namespace(){}
+
+const NamespaceElement Namespace::operator[] (int index) const{
+    return namespace_elements[index];
+}
+
+Namespace& Namespace::add_static_element(std::string value){
+    this->namespace_elements.push_back(NamespaceElement(value));
+    return *this;
+}
+
+Namespace& Namespace::add_dynamic_element(std::string name, std::string description ){
+    this->namespace_elements.push_back(NamespaceElement("*",name,description));
+    return *this;
+}
+
+std::vector<NamespaceElement> Namespace::get_namespace_elements() const{
+    return this->namespace_elements;
+}
+
+const bool Namespace::is_dynamic() const{
+    for(int i = 0 ; i < this->namespace_elements.size() ; i++ ){
+        if(this->namespace_elements[i].is_dynamic()){
+            return true;
+        }
+    }
+    return false;
+}
+
+const std::vector<int> Namespace::get_dynamic_indexes(){
+    std::vector<int> indexes;
+
+    for(int i = 0 ; i < this->namespace_elements.size() ; i++ ){
+        if(this->namespace_elements[i].is_dynamic()){
+            indexes.push_back(i);
+        }
+    }
+    return indexes;
+}
+
+NamespaceElement::NamespaceElement(std::string value, std::string name, std::string description)
+    :value(value)
+    ,name(name)
+    ,description(description){}
+
+NamespaceElement::NamespaceElement()
+    :value("")
+    ,name("")
+    ,description(""){}
+
+NamespaceElement::~NamespaceElement(){
+
+}
+void NamespaceElement::set_value(std::string v){
+    this->value = v;
+}
+void NamespaceElement::set_name(std::string n){
+    this->name = n;    
+}
+void NamespaceElement::set_description(std::string d){
+    this->description = d;
+}
+const std::string NamespaceElement::get_value() const {
```

Comment:
Please decide if you want to write `const{` or `const {` and use keep consistency.